av_transducer/calculators/asr_calculator.cc

```diff
@@ -0,0 +1,70 @@
+#include "mediapipe/framework/api2/node.h"
+#include "mediapipe/framework/api2/packet.h"
+#include "ml/asr/model.h"
+#include <vector>
+
+namespace aikit {
+
+class ASRCalculator : public mediapipe::api2::Node {
+public:
+    static constexpr mediapipe::api2::SideInput<std::string> kInASRModelPath{
+        "ASR_MODEL_PATH"};
+    static constexpr mediapipe::api2::SideInput<std::string> kInSPKModelPath{
+        "SPK_MODEL_PATH"};
+    static constexpr mediapipe::api2::SideInput<int> kInBufferDurationSec{
+        "BUFFER_DURATION_SEC"};
+    static constexpr mediapipe::api2::Input<std::vector<float>> kInAudio{"AUDIO"};
+    static constexpr mediapipe::api2::Output<aikit::ml::ASRResult>
+        kOutASRResult{"ASR_RESULT"};
+    MEDIAPIPE_NODE_CONTRACT(kInASRModelPath, kInSPKModelPath, kInBufferDurationSec, kInAudio, kOutASRResult);
+
+    absl::Status Open(mediapipe::CalculatorContext *cc) override;
+    absl::Status Process(mediapipe::CalculatorContext *cc) override;
+
+private:
+    std::unique_ptr<aikit::ml::ASRModel> model_;
+    std::vector<float> audio_buffer_;
+    size_t buffer_position_ = 0;
+    size_t buffer_size_;
+    static constexpr size_t kSampleRate = 16000;
+    static constexpr size_t kDefaultBufferDurationSec = 10;
+};
+
+MEDIAPIPE_REGISTER_NODE(ASRCalculator);
+
+absl::Status ASRCalculator::Open(mediapipe::CalculatorContext *cc) {
+    const std::string &asr_model_path = kInASRModelPath(cc).Get();
+    const std::string &spk_model_path = kInSPKModelPath(cc).Get();
+    model_ = std::make_unique<ml::ASRModel>(asr_model_path, spk_model_path);
+    int buffer_duration_sec = kDefaultBufferDurationSec;
+    if (cc->InputSidePackets().HasTag(kInBufferDurationSec.Tag())) {
```

Comment:
Note: Mediapipe has a new api now, where we can use fanier version to check that stream is connected.
Here is example: https://github.com/d61h6k4/call-assistant-bot/blob/main/av_transducer/calculators/ffmpeg_sink_video_calculator.cc#L97